programs/main.cpp

```diff
@@ -259,6 +261,10 @@ void writeError(const char * data)
         }
     }
 }
+/// Macros to avoid using strlen(), since it may fail if SSE is not supported.
+#define writeError(data) \
```

Comment:
There is an idiom to write `do {} while (0)` to avoid issues when someone will write something like `if (...) writeError(data);`. I'll correct it just in case...